src/items/capacitor.cpp

```diff
@@ -131,6 +132,36 @@ bool Capacitor::collectExtraInfo(QWidget * parent, const QString & family, const
 	return PaletteItem::collectExtraInfo(parent, family, prop, value, swappingEnabled, returnProp, returnValue, returnWidget, hide);
 }
 
+/**
+ * Sets the appropriate background colour in the combo box when the text is modified. When a user changes the text of an editable
+ * property in a combo box, this function is called. The function checks the validator of the combo box and sets a red background if the
+ * state is Intermediate (the value does is not within the limits of the property) or green if the state is Acceptable. Invalid states
+ * are not possible as the characters that make the string invalid are deleted if introduced.
+ * not changed in this function.
+ * @param[in] text The new string in the combo box.
+ */
+void Capacitor::textModified(const QString & text) {
+	FocusOutComboBox * focusOutComboBox = qobject_cast<FocusOutComboBox *>(sender());
+	if (focusOutComboBox == NULL) return;
+
+	int pos;
+	QString textTemp = text;
```

Comment:
I like the solution from ekhumoro :
https://stackoverflow.com/questions/58278508/get-visual-feedback-from-qvalidator

Instead of using the textModified signal, directly emit a validationChanged singal from our boundedregexpvalidator
This way the validation is only called once, and all the constness won't be an issue, too.

Besides, is currentTextChanged emitted if the text is not valid?

Reply:
Ok, I pushed a commit changing that. 
If the character introduced makes the text not valid, the character is not added to the current text. So, probably is not emitted, however the text has not changed either.